engine/src/main/java/org/hibernate/validator/internal/engine/MessageInterpolatorContext.java

```diff
@@ -37,17 +38,20 @@
 	private final Map<String, Object> messageParameters;
 	@Immutable
 	private final Map<String, Object> expressionVariables;
+	private final Path propertyPath;
 
 	public MessageInterpolatorContext(ConstraintDescriptor<?> constraintDescriptor,
-			Object validatedValue,
-			Class<?> rootBeanType,
-			Map<String, Object> messageParameters,
-			Map<String, Object> expressionVariables) {
+									  Object validatedValue,
+									  Class<?> rootBeanType,
+									  Map<String, Object> messageParameters,
+									  Map<String, Object> expressionVariables,
+									  Path propertyPath) {
```

Comment:
Let's not reindent things please.

And let's add `propertyPath` just after `rootBeanType`. It makes more sense.